frontend_tests/node_tests/filter.js

```diff
@@ -715,6 +717,50 @@ run_test('predicate_basics', () => {
         display_recipient: [{id: steve.user_id}, {id: me.user_id}],
     }));
     assert(!predicate({type: 'stream'}));
+
+    const img_msg = {
+        content: "<p><a href=\"\/user_uploads\/randompath\/test.jpeg\">test.jpeg<\/a><\/p>\n<div class=\"message_inline_image\"><a href=\"\/user_uploads\/randompath\/test.jpeg\" title=\"test.jpeg\"><img src=\"\/user_uploads\/randompath\/test.jpeg\"><\/a><\/div>",
+    };
+
+    const link_msg = {
+        content: "<p><a href=\"http:\/\/chat.zulip.org\">chat.zulip.org<\/a><\/p>",
+    };
+
+    const non_img_attachment_msg = {
+        content: "<p><a href=\"\/user_uploads\/randompath\/attachemnt.ext\">attachment.ext<\/a><\/p>",
+    };
+
+    const no_has_filter_matching_msg = {
+        content: "<p>Testing<\/p>",
+    };
+
+    predicate = get_predicate([['has', 'link']]);
+    let mockElem = $("<div>").append(link_msg.content);
+    mockElem.set_find_results("a:first", [$("<a>")]);
```

Comment:
I'm confused; how are these tests working? What is calling find on a `div` in the DOM to which you're appending these?

Reply:
Umm wait I will clean it up. I think I mixed up.